Refactored code:

ArunKumarJain opened the pull request:
Now tests don't need to set URLs instead it is moved to APIStub and APIStub is inheriting from MockServer

ArunKumarJain committed:
Refactored code:
Now tests don't need to set URLs instead it is moved to APIStub and APIStub is inheriting from MockServer

sivachandran commented on api_stub.py:
```diff
@@ -1,11 +1,13 @@
+from mockserver import MockServer
 
-class ApiStub():
+class ApiStub(MockServer):
```
Shouldn't it be called `ApiMock`?

sivachandran commented on api_stub.py:
```diff
@@ -25,7 +27,23 @@ def mobile_price(self, **kwargs):
         return json.dumps(priceJson)
 
-    def search(self, **kwargs):
+    def onMobileManufacturerRequest(self, response = None, **kwargs):
```
as discussed pass `response` through `kwargs`

sivachandran commented on api_stub.py:
```diff
@@ -1,11 +1,13 @@
-    def mobile_price(self, **kwargs):
+        super().__init__(host = host, port = port)
+
+    def onMobileModelInfo(self, **kwargs):
```
specify the route as doc comment for each `on` function

sivachandran commented on api_stub.py:
```diff
@@ -48,4 +66,36 @@ def bad_request(self, **kwargs):
\ No newline at end of file
+        return json.dumps(msg), code
+
+    def initialise(self):
```
why can't we do this under `__init__` func

sivachandran commented on tests.py:
```diff
@@ -1,46 +1,43 @@
+
         # adding little sleep as start & shutdown mock server very often creates
         # connection problems
         time.sleep(0.25)
```
we could do this inside `initialise` func

ArunKumarJain committed:
Few review comments addressed

ArunKumarJain committed:
Moved MockServer start inside initialise() func of ApiMock

ArunKumarJain committed:
Added decorator function to parse kwargs in callback functions
Renamed add_callback_response to add_response_callback
Renamed add_json_response to add_response_json

ArunKumarJain committed:
Fixed few comments and added new comments

sivachandran commented on mockserver.py:
```diff
@@ -1,6 +1,50 @@
+
+    def __call__(self, *args, **kwargs):
+
+        response = "{0}Response".format(self.func.__name__)
```
you could suffix `AttrName` to these variables for better readability.

ArunKumarJain committed:
Fixed review comments
